src/main/java/org/spectrumauctions/sats/opt/model/mrvm/MRVM_MIP.java

```diff
@@ -124,7 +124,7 @@ public Allocation adaptMIPResult(ISolution solution) {
                     double doubleQuantity = solution.getValue(xVar);
                     int quantity = (int) Math.round(doubleQuantity);
                     if (quantity > 0) {
-                        MRVMGenericDefinition def = new MRVMGenericDefinition(band, region);
+                        MRVMGenericDefinition def = world.getAllGenericDefinitions().stream().filter(g -> g.getBand().equals(band) && g.getRegion().equals(region)).findFirst().get();
```

Comment:
I agree that this makes sense, since we have this generic definition anyway in the world object. However, just to clarify: An `.equals(..)` comparison should not include the UUID by design (whether or not this is a good design is up to debate of course), as you can see in the class `GenericGood.java`:

```java

// ...

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public abstract class GenericGood implements SATSGood, Serializable {

    @Getter // <-- There is no "Include" from the Lombok library
    private final UUID uuid = UUID.randomUUID();

    @Getter
    @EqualsAndHashCode.Include // <-- There is one here
    @ToString.Include
    private final String name;

// ...
```
So I think here, the problem is simply that the next line "overrides" the previous MRVMGenericDefinition, so sneakily changes the UUID without noticing (since the HashCode stays the same).
Thus, your change definitely makes sense, we may want to think about a better design to prevent such situations in the first place, though.